bleach/utils.py

```diff
@@ -40,3 +43,52 @@ def force_unicode(text):
 
     # If not, convert it
     return six.text_type(text, "utf-8", "strict")
+
+
+netloc_port_re = re.compile(
+    "^" + URLValidator.netloc_re + URLValidator.port_re + "$", re.IGNORECASE
+)
+
+
+# Characters valid in scheme names
+scheme_chars = (
+    "abcdefghijklmnopqrstuvwxyz" "ABCDEFGHIJKLMNOPQRSTUVWXYZ" "0123456789" "+-."
+)
+
+
+def _is_valid_netloc_and_port(netloc):
+    """
+    Returns the scheme for a URI or None when parsing the URI fails
+
+    :arg str/unicode netloc:
+
+    :returns: bool
+
+    """
+    # The maximum length of a full host name is 253 characters per RFC 1034
+    # section 3.1. It's defined to be 255 bytes or less, but this includes
+    # one byte for the length of the name and one byte for the trailing dot
+    # that's used to indicate absolute names in DNS.
+    netloc = netloc_port_re.match(netloc)
+    return bool(netloc and len(netloc.group(0)) < 254)
```

Comment:
While this is mostly based on the Django code, I think this is worth adding tests for.

Reply:
Dropped the django direction.